framework/doc/content/syntax/Materials/index.md

```diff
@@ -1,4 +1,19 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
+# Materials System Overview
+
+Time derivative of a material property could be needed for transient calculations.
+MOOSE provides a function getMaterialPropertyDot that can be called in kernels, bcs, user objects, etc,
+which returns a reference of a time derivative of a real scalar material property.
+
+!alert note
+getMaterialPropertyDot currently only supports real scalar (Real) material property.
+
+!alert note
+getMaterialPropertyDot is not available in Materials.
```

Comment:
I will need to override this function in `Material.h` to throw an error when users are calling this in `Material`. I did not realize `Material` is also derived from `MaterialPropertyInterface`.